src/main/java/org/springframework/integration/aws/outbound/KplMessageHandler.java

```diff
@@ -115,8 +120,29 @@ public void setConverter(Converter<Object, byte[]> converter) {
 		setMessageConverter(new ConvertingFromMessageConverter(converter));
 	}
 
+	/**
+	 * When in KPL mode, the setting allows handling backpressure on the KPL native process. Setting this value would enable a sleep on the KPL Thread for the specified number of milliseconds defined in maxRecordInFlightsSleepDurationInMillis.
+	 *
+	 * @param maxRecordsInFlight Defaulted to 0. Value of 0 indicates that Backpressure handling is not enabled. Specify a positive value to enable back pressure.
+	 */
+	public void setMaxOutstandingRecordsInFlight(long maxRecordsInFlight) {
+		Assert.isTrue(maxRecordsInFlight > 0, "'maxRecordsInFlight must be greater than 0.");
+		this.maxRecordsInFlight = maxRecordsInFlight;
+	}
+
+	/**
+	 * The setting allows handling backpressure on the KPL native process. Enabled when maxOutstandingRecordsCount is greater than 0. The configurations puts the KPL Thread to sleep for the specified number of milliseconds.
+	 *
+	 * @param maxRecordInFlightsSleepDurationInMillis Default is 100ms.
+	 */
+	public void setMaxRecordInFlightsSleepDurationInMillis(int maxRecordInFlightsSleepDurationInMillis) {
```

Comment:
DITTO here: `setMaxInFlightRecordsDuration`.
With respective code style for length and `@since 3.0.9`